Install a Swift 5.8.1 toolchain in all swift-ci images

bnbarham opened the pull request:
Installs a Swift 5.8.1 toolchain into `/opt/swift/<version>` on each Linux platform. Removes clang since the toolchain clang can be used instead.

DougGregor committed:
Install a Swift 5.8.1 toolchain in all swift-ci images

Installs a Swift 5.8.1 toolchain into `/opt/swift/<version>` on each
Linux platform. Removes clang since the toolchain clang can be used
instead.

DougGregor reviewed (approved):
There's no way to share this code across Dockerfiles?

tomerd commented on swift-ci/master/amazon-linux/2/Dockerfile:
```diff
@@ -47,6 +46,51 @@ RUN mkdir -p /usr/local/lib/python3.7/site-packages/
+    && rm -rf "$GNUPGHOME" swift.tar.gz.sig swift.tar.gz
+
+ENV PATH="${SWIFT_PREFIX}/usr/bin:${PATH}"
+
```
should we use the RPM packages instead?

MaxDesiatov replied:
We currently mark those as "experimental use only" on swift.org/download. Are we willing to try those out here anyway, or is description on that page outdated?

MaxDesiatov commented on swift-ci/master/amazon-linux/2/Dockerfile:
```diff
@@ -2,7 +2,7 @@ FROM amazonlinux:2
 RUN yum install shadow-utils -y
 
-RUN groupadd -g 998 build-user && \
+RUN groupadd -g 1000 build-user && \
```
I don't have any opinion on the group number, just curious why this change was made?

bnbarham replied:
I cannot remember why this was needed. It definitely was for CentOS when using the release image, but I don't know if it is still needed now that we're not. I'm guessing probably not? I don't suppose you remember @shahmishal / @DougGregor?

DougGregor replied:
I have no memory of this place

bnbarham replied:
I've reverted for now. We'll see if there's any errors pretty quick.

shahmishal committed:
Use swift.org keys to avoid timeout and revert group id change

bnbarham committed:
Add missing python3-pkg-resources dependencies

bnbarham committed:
Use swift.org keys to avoid timeout on the rest of the platforms

This was only changed for Ubuntu, also update for Amazon Linux, CentOS,
and UBI.

bnbarham committed:
Add diffutils to UBI 9

This presumably came through clang previously as a transitive
dependency. Add it directly.
